test/e2e/utils_test.go

```diff
@@ -204,6 +205,77 @@ func verifyDeploymentArgs(k8sclient *kubernetes.Clientset, deploymentName string
 	})
 }
 
+// addOverrideResources adds the override resources to specific the cert-manager operand. The update process
+// is retried if a conflict error is encountered.
+func addOverrideResources(client *certmanoperatorclient.Clientset, deploymentName string, res v1alpha1.CertManagerResourceRequirements) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		operator, err := client.OperatorV1alpha1().CertManagers().Get(context.TODO(), "cluster", v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		updatedOperator := operator.DeepCopy()
+
+		switch deploymentName {
+		case certmanagerControllerDeployment:
+			updatedOperator.Spec.ControllerConfig = &v1alpha1.DeploymentConfig{
+				OverrideResources: res,
+			}
+		case certmanagerWebhookDeployment:
+			updatedOperator.Spec.WebhookConfig = &v1alpha1.DeploymentConfig{
+				OverrideResources: res,
+			}
+		case certmanagerCAinjectorDeployment:
+			updatedOperator.Spec.CAInjectorConfig = &v1alpha1.DeploymentConfig{
+				OverrideResources: res,
+			}
+		default:
+			return fmt.Errorf("unsupported deployment name: %s", deploymentName)
+		}
+
+		_, err = client.OperatorV1alpha1().CertManagers().Update(context.TODO(), updatedOperator, v1.UpdateOptions{})
+		return err
+	})
+}
+
+// verifyDeploymentResources polls every 1 second to check if the deployment resources is updated to contain
+// the passed resources. It returns an error if a timeout (5 mins) occurs or an error was encountered while
+// polling the deployment resources.
+func verifyDeploymentResources(k8sclient *kubernetes.Clientset, deploymentName string, res v1alpha1.CertManagerResourceRequirements, added bool) error {
+
+	return wait.PollImmediate(time.Second*5, time.Minute*5, func() (done bool, err error) {
+		controllerDeployment, err := k8sclient.AppsV1().Deployments(operandNamespace).Get(context.TODO(), deploymentName, v1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
```

Comment:
1. Your comment is "polls every 1 second" while argument is time.Second*5. I suggest 10 seconds.
2. Is it needed to check `IsNotFound` specially? Does the test have chance to see `IsNotFound`?